### PR TITLE
Fix SonarQube issues

### PR DESCRIPTION
**Pull request by harry1453**
This fixes bugs and vulnerabilities picked up by sonarqube.

https://sonarcloud.io/dashboard?id=burst-apps-team_burstcoin

### CODE REVIEW

File: src/brs/Block.java [hunk 1]
```diff
@@ -31,18 +33,18 @@
   private final int payloadLength;
   private final byte[] generationSignature;
   private final byte[] payloadHash;
-  private volatile List<Transaction> blockTransactions;
+  private final AtomicReference<List<Transaction>> blockTransactions = new AtomicReference<List<Transaction>>();
```
Comment (ohager) [inline src/brs/Block.java]: 👍

File: src/brs/peer/Peers.java [hunk 2]
```diff
@@ -471,6 +472,7 @@ public void run() {
             }
           }
           catch (InterruptedException ex) {
+            Thread.currentThread().interrupt();
```
Comment (ohager) [inline src/brs/peer/Peers.java]: 👍

File: src/brs/http/APIServlet.java [hunk 3]
```diff
@@ -236,13 +236,23 @@ void addErrorMessage(HttpServletResponse resp, JsonElement msg) throws IOExcepti
   private static Map<String, PrimitiveRequestHandler> primitiveRequestHandlers;
 
   @Override
-  protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
-    process(req, resp);
+  protected void doGet(HttpServletRequest req, HttpServletResponse resp) {
+    try {
+      process(req, resp);
+    } catch (Throwable t) { // We don't want to send exception information to client...
+      resp.setStatus(500);
```
Comment (blankey1337) [inline src/brs/http/APIServlet.java]: Is it safe al ways throw a 500 here? thinking about 4xx errors too.

**Comment by blankey1337** [comment]
Very nice, always a good idea to use pure functions over mutation.

File: src/brs/http/APIServlet.java (hunk 3 above)
Comment (harry1453) [reply src/brs/http/APIServlet.java]: The implementation of the process method *should* handle any expected errors - if an exception was not caught by it we should probably throw a 500.

**Comment by harry1453** [comment]
The problem is that when you make an object volatile, you actually just make the pointer to the object volatile, not the object itself, which is why you should use atomics

**Comment by harry1453** [comment]
There is one issue left complaining about readability which I am not going to deal with now as it risks breaking stuff.

**Pull request merged**